src/app/[locale]/_components/QrScanner.tsx

```diff
@@ -38,62 +36,70 @@ export function QrScanner(props: { onScan: (scanResult: string) => Promise<void>
             });
             setState(State.Stopped);
         }
-    };
+    }, []);
 
-    const handleScan = async (result: string) => {
-        setState(State.HandleQr);
-        try {
-            await props.onScan(result);
-            setState(State.Stopped);
-        } catch (e) {
-            const msg = e instanceof LocalizedError ? e.descriptor : messages.mnestix.qrScanner.defaultCallbackErrorMsg;
-            notificationSpawner.spawn({
-                message: intl.formatMessage(msg),
-                severity: 'error',
-            });
-            setState(State.LoadScanner);
-        }
-    };
+    const handleScan = useCallback(
+        async (result: string) => {
+            setState(State.HandleQr);
+            try {
+                await props.onScan(result);
+                setState(State.Stopped);
+            } catch (e) {
+                const msg =
+                    e instanceof LocalizedError ? e.descriptor : messages.mnestix.qrScanner.defaultCallbackErrorMsg;
+                notificationSpawner.spawn({
+                    message: intl.formatMessage(msg),
+                    severity: 'error',
+                });
+                setState(State.LoadScanner);
+            }
+        },
+        [props.onScan],
+    );
+
+    if (typeof window !== 'undefined' && typeof window.Cypress !== 'undefined') {
+        window.Cypress.scannerCallback = handleScan;
+    }
```

Comment:
I think this is super ugly tbh, but did not find any other way.
We cannot access the handleScan function directly, as it is a local variable in the function component.
We cannot export the function, as it uses a state from the component.
We should not make it a class component, because Next.js and general consistency

Reply:
Can you add a comment explaining why we have this? Looks like something which might confuse us in a few weeks 😄